src/providers/coinbase/coinbase.ts

```diff
@@ -761,4 +767,53 @@ export class CoinbaseProvider {
       });
     });
   }
+
+  public payInvoice(invoiceId, currency, twoFactorCode?) {
+    return new Promise((resolve, reject) => {
+      const url = 'https://bitpay.com/oauth/coinbase/pay/' + invoiceId;
+      const data = {
+        currency,
+        token: this.accessToken,
```

Comment:
Before send `token` you should check that it isn't expired.

Reply:
the only way to know if the token expired is with the error. So as a solution now the token is refreshed before trying to make the payment